### PR TITLE
HDTorrents

### PR DESCRIPTION
**Pull request by damwthomas**

### CODE REVIEW

**Comment by zone117x** [comment]
Looks like a lot of work, nice job. We definitely need these features. Can you fix the merge conflicts? You need to pull latest into your fork and update your pull request.

**Comment by damwthomas** [comment]
I think it went well, I hope I didn't fuck up :D

**Comment by zone117x** [comment]
Whats left to be done with the auth feature?

**Comment by ghost** [comment]
Could you hash the password rather than storing it plain text?

**Comment by damwthomas** [comment]
If i'm not mistaken the password is base64 encoded (i know, it's as safe as storing it in plain text). But i started of with md5 hashing it but then obviously i wasn't able to change my password. The problem atm is that sometimes when you delete your password it keeps asking for it until jackett has been restarted. Or when you login with wrong data, then you give in the correct data that the user has to manually refresh the page or that the js an css aren't loaded properly. This has been less an issue since i added a custom header to the auth request but still, i'd like it more polished before releasing it.

**Comment by damwthomas** [comment]
This can be merged, HDtorrents added, some classes for the authentication (that will come later)

File: src/Jackett/Indexers/HDTorrents.cs [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+                            release.Seeders = seeders;
+                        if (int.TryParse(qRow.Find("td").Get(10).FirstChild.FirstChild.InnerText, out peers))
+                            release.Peers = peers;
+
+                        string fullSize = qRow.Find("td.mainblockcontent").Get(6).InnerText;
+                        string[] sizeSplit = fullSize.Split(' ');
+                        switch (sizeSplit[1].ToLower())
+                        {
+                            case "kb":
+                                size = ReleaseInfo.BytesFromKB(float.Parse(sizeSplit[0], CultureInfo.InvariantCulture));
```
Comment (zone117x) [inline src/Jackett/Indexers/HDTorrents.cs]: Can you update all float.Parse, int.Parse, etc.. to use ParseUtil.Coerce..? I just implemented it yesterday to deal with all the number parsing bug reports we were getting. 
https://github.com/zone117x/Jackett/blob/master/src/Jackett/ParseUtil.cs

Comment (damwthomas) [inline src/Jackett/Indexers/HDTorrents.cs]: OH yes, I'm sorry

File: src/Jackett/Security.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+﻿using System;
```
Comment (zone117x) [inline src/Jackett/Security.cs]: Can you not include the classes unrelated to HDTorrents in this pull request? Not to be a pain the ass - just gotta make sure we can review the feature thoroughly once you've completed it.

**Comment by zone117x** [comment]
@damwthomas Can you get back on IRC to chat? 
EDIT: Didn't mean to press the close button for this comment.

**Comment by zone117x** [comment]
Perfect! Thanks

**Pull request merged**